src/Services/PasswordHashingService.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Blumilk\Meetup\Core\Services;
+
+use Blumilk\Meetup\Core\Models\User;
+use Illuminate\Contracts\Hashing\Hasher;
+
+class PasswordHashingService
+{
+    public function __construct(
+        protected Hasher $hasher,
+    ) {}
+
+    public function hash(User $user): void
+    {
+        if (!$this->hasher->info($user->password)["algo"]) {
+            $user->password = $this->hasher->make($user->password);
+        }
```

Comment:
For what is this `if` statement?

Reply:
checks if the password is already hashed

Reply:
I'm not sure that it's a good way. Why do you need to check if password is hashed already if user sends plain text password in the request?
There are 3 places where password should be hashed - create new user, update user, reset password. All are related with User model.
Maybe it will be better to hash password in the mutator? It will always hash password before create/update User model.

https://laravel.com/docs/9.x/eloquent-mutators#defining-a-mutator

```php
protected function password(): Attribute
    {
        return Attribute::make(
            set: fn (string $value): string => Hash::make($value),
        );
    }
```

Reply:
I think this is not a bad idea, but for now I will not use it and I will stay with the service as suggested by @krzysztofrewak